prio/rand.c

```diff
@@ -16,18 +16,22 @@
 #include "rand.h"
 #include "util.h"
 
+#include "callbk.h"
+
 #define CHUNK_SIZE 8192
 
 static NSSInitContext* prioGlobalContext = NULL;
 
 SECStatus
-rand_init(void)
+rand_init(nss_struct *nss_pointer)
```

Comment:
This approach assumes that rand_init is not invoked in parallel. Did you confirm if this is the case, or if the API allows parallel calls to rand_init? If it doesn't this is fine, if it does, you need a lock somewhere.

Reply:
I looked in prio(pclient) and tests, there are no parallel calls.